grakn-graql/src/main/java/ai/grakn/graql/internal/printer/GraqlPrinter.java

```diff
@@ -90,15 +91,17 @@ public String build(Function<StringBuilder, StringBuilder> builder) {
             }
 
             if (concept.isRelation()) {
-                String relationString = concept.asRelation().rolePlayers().entrySet().stream().map(entry -> {
+                String relationString = concept.asRelation().allRolePlayers().entrySet().stream().map(entry -> {
                     RoleType roleType = entry.getKey();
-                    Instance rolePlayer = entry.getValue();
+                    Set<Instance> instances = entry.getValue();
 
-                    if (rolePlayer != null) {
-                        String s = colorType(roleType) + ": id " + idToString(rolePlayer.getId());
-                        return Optional.of(s);
-                    } else {
+                    if(instances.isEmpty()){
                         return Optional.<String>empty();
+                    } else {
+                        StringBuilder innerSb = new StringBuilder();
+                        instances.forEach(rolePlayer ->
+                                innerSb.append(colorType(roleType)).append(": id ").append(idToString(rolePlayer.getId())).append(","));
+                        return Optional.of(innerSb.toString());
```

Comment:
This one is interesting. What would help here (and elsewhere) is a `Collection<Pair<RoleType, Instance>>`, which the API doesn't provide. I'm not advocating adding it now, but it's something to think about!